.cirrus.yml

```diff
@@ -9,49 +9,46 @@ task:
     dockerfile: .ci/Dockerfile
   env:
     INTEGRATION_TEST_PATH: "./packages/integration_test"
-  upgrade_script:
-    - flutter channel stable
-    - flutter upgrade
-    - flutter channel master
-    - flutter upgrade
-    - flutter config --enable-linux-desktop
-    - git fetch origin master
+  setup_script:
+    - git fetch origin master # To set FETCH_HEAD for "git merge-base" to work
   matrix:
     ### Platform-agnostic tasks ###
     - name: plugin_tools_tests
+      upgrade_script: ./script/set_channel.sh master
```

Comment:
You could pull this up to the `task` level instead of making each named task redeclare it:
```
  setup_script:
    - git fetch origin master # To set FETCH_HEAD for "git merge-base" to work
   upgrade_script: ./script/set_channel.sh $CHANNEL
```
and then set `CHANNEL` in the few places missing it.
```
matrix:
    ### Platform-agnostic tasks ###
    - name: plugin_tools_tests 
    - env:
        CHANNEL: "master"
```
etc

Reply:
Or you can do it the way I tried in 2019 and set CHANNEL to `master` at the top level, and let individual tests override it
https://github.com/jmagman/plugins/blob/41147b069a15a6199b27a4becfcd6d4cdf5b502c/.cirrus.yml#L13-L15
```yaml
  environment:
    CHANNEL: "master" # Default to flutter master, tests can override.
```

Reply:
For some reason I assumed I couldn't use env variables "above" the level in the tree they are set, but in retrospect I'm not sure why I would have thought that. Will reworking to avoid the script.